module/mempool/herocache/backdata/heropool/pool_test.go

```diff
@@ -215,6 +217,111 @@ func TestInvalidateEntity(t *testing.T) {
 	}
 }
 
+// TestAddAndRemoveEntities checks health of heroPool for scenario where entitites are stored and removed in a predetermined order.
+// LRUEjection, NoEjection and RandomEjection are tested. RandomEjection doesn't allow to provide a final state of the pool to check.
+func TestAddAndRemoveEntities(t *testing.T) {
+	for _, tc := range []struct {
+		limit               uint32       // capacity of the pool
+		entityCount         uint32       // total entities to be stored
+		ejectionMode        EjectionMode // ejection mode
+		numberOfOperations  int
+		probabilityOfAdding float32
+	}{
+		{
+			limit:               500,
+			entityCount:         1000,
+			ejectionMode:        LRUEjection,
+			numberOfOperations:  1000,
+			probabilityOfAdding: 0.8,
+		},
+		{
+			limit:               500,
+			entityCount:         1000,
+			ejectionMode:        NoEjection,
+			numberOfOperations:  1000,
+			probabilityOfAdding: 0.8,
+		},
+		{
+			limit:               500,
+			entityCount:         1000,
+			ejectionMode:        RandomEjection,
+			numberOfOperations:  1000,
+			probabilityOfAdding: 0.8,
+		},
+	} {
+		t.Run(fmt.Sprintf("%d-limit-%d-entities", tc.limit, tc.entityCount), func(t *testing.T) {
+			testAddRemoveEntities(t, tc.limit, tc.entityCount, tc.ejectionMode, tc.numberOfOperations, tc.probabilityOfAdding)
+		})
+	}
+}
+
+// Indexes contains an index in the pool of an inserted entity and an index of this entity
+// in the array of entities created by the EntityListFixture.
+type Indexes struct {
+	indexInPool        EIndex
+	indexInEntitiesArr int
+}
```

Comment:
Given that this test package isn't isolated from the primary `pool` code package, it is advisable not to define unique structs solely for testing purposes. These structs have the potential to bleed into the core logic, thereby complicating the maintainability of the codebase. To foster simplicity and avoid this issue, please employ native Go data structures during testing.

For instance, the current structure can be replaced by two separate maps. The first map would maintain the index within the pool, while the second would keep track of the index within the entities array. Such a modification would align with the principle of simplicity, making the tests easier to comprehend and the codebase more maintainable.

Reply:
It's a good point. But I would approach it differently - we could just rename this struct from Indexes to PoolIndexAndIndexInArrayOfEntities.

2 maps aren't efficient and are harder to maintain. 

I'll implement you suggestion with 2 maps, but please consider my proposal.

Reply:
sure, implemented